File: src/ml/Models/StatisticsCategorical.py

import json

class StatisticsCategorical:
    
    def __init__(
        self,
        valid_count  = 0,
        na_count     = 0,
        unique_count = 0,
        most_common  = 0,
        frequencies  = []
        ) -> None:
        
        self.ValidCount  = int(valid_count)
        self.NaCount     = int(na_count)
        self.UniqueCount = int(unique_count)
        self.MostCommon  = (str(most_common[0]), str(float(most_common[1])))
        self.Frequencies = [(str(a), str(float(b))) for (a, b) in frequencies]

File: src/backend/dotNet/dotNet/Models/StatisticsCategorical.cs
namespace dotNet.Models {
    public class StatisticsCategorical {
        public StatisticsCategorical(
            int validCount, int naCount, int uniqueCount, 
            KeyValuePair<string, float> mostCommon, 
            List<KeyValuePair<string, float>>? frequencies) {

            ValidCount = validCount;
            NaCount = naCount;
            UniqueCount = uniqueCount;
            MostCommon = mostCommon;
            Frequencies = frequencies;
        }

        public static Dictionary<string, StatisticsCategorical>? Load(string data) {
            return Newtonsoft.Json.JsonConvert.DeserializeObject<Dictionary<string, StatisticsCategorical>>(data);
        }

        public int ValidCount { get; set; }
        public int NaCount { get; set; }
        public int UniqueCount { get; set; }
        public KeyValuePair<string, float> MostCommon { get; set; }
        public List<KeyValuePair<string, float>>? Frequencies { get; set; }
    }
}
